#!/usr/bin/env python
PACKAGE = "connectivity_controller"
#import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k", double_t, 0, "Translation gain", .2, -1.5, 1.5)

gen.add("ktheta", double_t, 0, "Rotation gain", 1.0, -15, 15)

gen.add("power", bool_t, 0, "ON/OFF", True)

gen.add("x_bar_desired", double_t, 0, "Desired x bar", 0.25, -3, 3)

gen.add("y_bar_desired", double_t, 0, "Desired y bar", 0.5, -3, 3)

gen.add("psi_desired", double_t, 0, "Desired psi", 2.35619449019, -6.28318530718, 6.28318530718)

gen.add("k_bar", double_t, 0, "u bar gain", 1.0, -5, 5)

gen.add("kconn", double_t, 0, "Connectivity gain", 0.1, -2, 2)

gen.add("kca", double_t, 0, "Collision Avoidance gain", 0.08, -0.2, 0.2)

gen.add("kfc", double_t, 0, "Formation Control gain", 1.0, -5, 5)

gen.add("formation", bool_t, 0, "Formation control and collision avoidance (true) or u bar (false)", False)

exit(gen.generate(PACKAGE, "connectivity_controller", "ControlVariables"))
